index.js

```diff
@@ -191,6 +193,21 @@ class MemoryCacheStore {
       entries = []
       pathValues.set(key.method, entries)
     }
+
+    const existingEntry = entries.findIndex((entry) => (
+      (entry.vary == null || Object.keys(entry.vary).every(headerName => {
+        if (entry.vary[headerName] === null) {
+          return key.headers[headerName] === undefined
+        }
+
+        return entry.vary[headerName] === key.headers[headerName]
+      }))
+    ))
+    if (existingEntry >= 0) {
+      entries.splice(existingEntry, 1, entry)
```

Comment:
I don't think that splice is a correct way to do that because it skip all related entities invalidation logic. For example a new cache entry might have the same url and vary headers, but different tags. What I would do is to add something like that before saving a new entity:

```js
const existingEntity = this.get(key)
if (existingEntity) {
  this.#deleteByKey(key)
}

// all other insert logic
```

Reply:
done